Sources/MarathonCore/Export.swift

```diff
@@ -0,0 +1,174 @@
+                printer.output("❓  Are you sure you want to overwrite it? (Type 'Y' to confirm)")
+
+                if readLine()?.lowercased() == "y" {
+                    try exportFolder.subfolder(atPath: scriptName).delete()
+                } else {
+                    exit(1)
+                }
+            }
+        }
+
+        try perform(export(file, to: exportFolder), orThrow: Error.failedToExportScript(file.name))
+    }
+
+    private func export(_ file: File, to folder: Folder) throws {
+        let projectFolder = try folder.createSubfolder(named: file.name.asScriptName())
+        let sourcesFolder = try projectFolder.createSubfolder(named: "Sources")
+        let scriptFileData = try file.read()
+        try sourcesFolder.createFile(named: file.name, contents: scriptFileData)
+
+        let packageFile = try projectFolder.createFile(named: "Package.swift")
+        let packages = try resolvePackages(from: file)
+        let packageFileString =  try makePackageFileString(for: file, with: packages)
+        try packageFile.write(string: packageFileString)
+    }
+
+    private func resolvePackages(from file: File) throws -> [Package] {
+        let importLines = try file.readAsString()
+            .components(separatedBy: .newlines)
+            .filter { $0.hasPrefix("import") }
+        let scriptDependencyNames = importLines
+            .flatMap { $0.components(separatedBy: .whitespaces) }
+            .filter { !$0.hasPrefix("import") && !$0.hasPrefix("//") && !$0.hasPrefix("marathon") }
+
+        let script = try scriptManager.script(at: file.path)
+        let description = try packageManager.makePackageDescription(for: script)
+
+        let managedPackages = description
+            .components(separatedBy: .newlines)
+            .filter { $0.contains(".Package") }
+            .map { $0.trimmingCharacters(in: .whitespaces) }
+            .map { $0.replacingOccurrences(of: ".Package", with: "") }
+            .map { $0.replacingOccurrences(of: "(", with: "") }
+            .map { $0.replacingOccurrences(of: ")", with: "") }
+            .map { $0.replacingOccurrences(of: "url:", with: "") }
+            .map { $0.replacingOccurrences(of: " majorVersion: ", with: "") }
+            .map { $0.replacingOccurrences(of: "\"", with: "") }
+            .map { $0.replacingOccurrences(of: " ", with: "") }
+
+        var packages: [Package] = []
+        for name in scriptDependencyNames {
+            guard let match = managedPackages.first(where: { $0.contains(name) }) else {
+                continue
+            }
+
+            let components = match.components(separatedBy: ",").filter { !$0.isEmpty }
+            guard let url = URL(string: components[0]), let majorVersion = Int(components[1]) else {
+                continue
+            }
+            let package = Package.init(name: name, url: url, majorVersion: majorVersion)
+            packages.append(package)
+        }
+        return packages
+    }
```

Comment:
This seems to work but, it still feels super dirty. I went ahead and did the work inside of Export.swift for now since theres no reason it needs to be inside of script manager expect for code duplication. I think this method of resolving the script's Package file is closer to  what the end goal might look like. I understand now that marathon maintains one list of all the packages it maintains for all its scripts and what I need to do is look at the import statements of the script to find out which if any of those managed packages the script importing.